squidDrone/src/i2c/i2c_interface.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef SRC_I2C_INTERFACE_HPP_
+#define SRC_I2C_INTERFACE_HPP_
+
+#include "i2c_status.hpp"
+#include <stdint.h>
+#include <vector>
+#include <tuple>
+
+namespace i2c {
+
+  /**
+   * @brief Interface of I2C package
+   * 
+   */
+  class I2CInterface {
+    public:
+      virtual ~I2CInterface() = default;
+      explicit I2CInterface(void){};
+
+      /**
+       * @brief Reads data from I2C Bus
+       * 
+       * @param address The address of the I2C Bus participant
+       * @param byte_size Amount of Bytes to read
+       * @param timeout Timeout in milliseconds
+       * @return std::tuple<I2CStatus, std::vector<uint8_t>> Status of I2C Interface and Data read as std::vector. Each element is a Byte read.
```

Comment:
please be more specifi

Reply:
I appended the documentation in this commit: https://github.com/elheck/Squiddrone/pull/219/commits/29856cf1f262ef9c48fe6a87d1780ae11a7daf76